osquery/tables/system/windows/certificates.cpp

```diff
@@ -0,0 +1,296 @@
+                                    &(certContext->pCertInfo->Issuer),
+                                    CERT_SIMPLE_NAME_STR,
+                                    nullptr,
+                                    0);
+    certBuff.resize(issuerSize);
+    std::fill(certBuff.begin(), certBuff.end(), 0);
+    issuerSize = CertNameToStr(certContext->dwCertEncodingType,
+                               &(certContext->pCertInfo->Issuer),
+                               CERT_SIMPLE_NAME_STR,
+                               certBuff.data(),
+                               issuerSize);
+    r["issuer"] = issuerSize == 0 ? "" : std::string(certBuff.data());
+
+    // TODO: Find the right API calls to get whether a cert is for a CA
+    r["ca"] = INTEGER(-1);
+
+    r["self_signed"] =
+        WTHelperCertIsSelfSigned(CERT_ENCODING, certContext->pCertInfo)
+            ? INTEGER(1)
+            : INTEGER(0);
+
+    r["not_valid_before"] =
+        std::to_string(filetimeToUnixtime(certContext->pCertInfo->NotBefore));
+
+    r["not_valid_after"] =
+        std::to_string(filetimeToUnixtime(certContext->pCertInfo->NotAfter));
+
+    r["signing_algorithm"] =
+        cryptOIDToString(certContext->pCertInfo->SignatureAlgorithm.pszObjId);
+
+    r["key_algorithm"] = cryptOIDToString(
+        certContext->pCertInfo->SubjectPublicKeyInfo.Algorithm.pszObjId);
+
+    r["key_usage"] = getKeyUsage(certContext->pCertInfo);
+
+    r["key_strength"] = INTEGER(certContext->cbCertEncoded);
+
+    std::stringstream subjId;
+    // TODO: Sort out why we're not getting back Subject IDs
+    if (certContext->pCertInfo->SubjectUniqueId.cbData > 0) {
+      try {
+        boost::algorithm::hex(certContext->pCertInfo->SubjectUniqueId.pbData,
+                              std::ostream_iterator<char>{subjId, ""});
+      } catch (std::exception /* e */) {
+      }
+      r["subject_key_id"] = subjId.str();
+    } else {
+      r["subject_key_id"] = "-1";
+    }
+
+    std::stringstream issuerId;
+    // TODO: Sort out why we're not getting back Issuer IDs
+    if (certContext->pCertInfo->IssuerUniqueId.cbData > 0) {
+      boost::algorithm::hex(certContext->pCertInfo->IssuerUniqueId.pbData,
+                            std::ostream_iterator<char>{issuerId, ""});
+      r["authority_key_id"] = issuerId.str();
+    } else {
+      r["authority_key_id"] = "-1";
+    }
+
+    r["path"] = certStoreName.empty() ? "-1" : certStoreName;
+    results.push_back(r);
+    certContext = CertEnumCertificatesInStore(certStore, certContext);
+  }
+}
+
+/// Windows API callback for processing a system cert store
+BOOL WINAPI certEnumSystemStoreCallback(const void* systemStore,
+                                        unsigned long flags,
+                                        PCERT_SYSTEM_STORE_INFO storeInfo,
+                                        void* reserved,
+                                        void* arg) {
+  auto storeArg = *static_cast<ENUM_ARG*>(arg);
```

Comment:
Why dereference the `static_cast`, maybe hold the `auto` as a `auto*`. Keep it as close to reality until you need to use it.